cubequery/tasks/__init__.py

```diff
@@ -367,6 +369,51 @@ def validate_standard_spatial_query(aoi, countries):
     return errors
 
 
+def check_database_spatial(aoi, platform, start_date, end_date):
+    # Check that the params are safe and valid
+    if not aoi.startswith("POLYGON"):
+        raise ValueError("AOI must be a polygon")
+
+    # Check if start and end date are strings of dates
+    if datetime.strptime(start_date, "%Y-%m-%d") is None:
+        raise ValueError("Start date must be a valid date")
+
+    if datetime.strptime(end_date, "%Y-%m-%d") is None:
+        raise ValueError("End date must be a valid date")
+
+    # Connect to database
+    connection = psycopg2.connect(host=get_config("Database", "host"),

```

Comment:
Probably should be cached in the flask app but this will work with the amount of load we have for now.